group-membership/src/Code.js

```diff
@@ -0,0 +1,151 @@
+const EMAIL = 'Email';
+const GOOGLE_GROUP = 'Google Group';
+const ALLOWED = 'Allowed';
+const EMAIL_TEMPLATE_DOC_URL = 'Email template doc URL';
+const EMAIL_SUBJECT = 'Email subject';
+const STATUS = 'Status';
+
+const STATUS_VALUE = {
+  sent: 'Sent',
+  alreadyInGroup: 'Already in group',
+  notSent: 'Not sent',
+  requiredFieldMissing: 'Required field(s) missing: fill out all fields for this row',
+  emptyRow: '',
+};
+
+/**
+ * Installs a trigger in the Spreadsheet to run upon the Sheet being opened.
+ * To learn more about triggers read:
+ * https://developers.google.com/apps-script/guides/triggers
+ */
+function onOpen() {
+  SpreadsheetApp.getUi()
+      .createMenu('Install trigger')
+      .addItem('onEdit', 'installOnEditTrigger')
+      .addToUi();
+}
+
+/**
+ * Installs a trigger in the Spreadsheet that is scheduled
+ * to run upon when values in the Sheet are edited.
+ * To learn more about triggers read:
+ * https://developers.google.com/apps-script/guides/triggers/installable
+ */
+function installOnEditTrigger() {
+  ScriptApp.newTrigger('onEditInstallableTrigger')
+      .forSpreadsheet(SpreadsheetApp.getActive())
+      .onEdit()
+      .create();
+}
+
+/**
+ * Trigger that runs on edit after being installed via the interface.
+ *
+ * @param {Object} e - onEdit trigger event.
+ */
+function onEditInstallableTrigger(e) {
+  // Get the headers, row range and values from the active sheet.
+  const sheet = SpreadsheetApp.getActiveSheet();
+  const headers = sheet.getDataRange().offset(0, 0, 1).getValues()[0];
+  const range = sheet.getRange(e.range.getRow(), 1, 1, headers.length);
+  const row = range.getValues()[0];
+
+  // Convert the row Array into an entries Object using the headers for the
+  // field names.
+  const entries = headers.reduce((result, columnName, i) => {
+    result[columnName] = row[i];
+    return result;
+  }, {});
+
+  // Update the entries Object with the status returned by addToGroup().
+  try {
+    const statusValue = addToGroup(
+        entries[EMAIL],
+        entries[GOOGLE_GROUP],
+        entries[ALLOWED],
+        entries[EMAIL_TEMPLATE_DOC_URL],
+        entries[EMAIL_SUBJECT]
+    );
+    entries[STATUS] = statusValue == STATUS_VALUE.sent ?
+        `${statusValue}: ${new Date()}` : statusValue;
+  } catch (e) {
+    // If there's an error, report that as the status.
+    entries[STATUS] = e;
+  }
+
+  // Convert the updated entries Object into a row Array.
```

Comment:
Don't really need a self-explanatory comment

Reply:
I don't think it hurts to be a bit more explicit. I would like to keep the comment.

Reply:
In the future I'd like to keep the style across all the solutions consistent so it's easier to maintain the whole body of work